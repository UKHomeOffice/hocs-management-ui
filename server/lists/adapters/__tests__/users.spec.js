const usersAdapter = require('../users');

const mockLogger = {
    debug: () => { },
    info: () => { },
    warn: () => { },
    error: () => { }
};

describe('User Adapter', () => {

    it('should transform and sort user data', async () => {
        const mockData = [
            { firstName: 'User', lastName: 'A', email: 'user.a@example.com', id: 1 },
            { firstName: 'User', lastName: 'C', email: 'user.c@example.com', id: 3 },
            { firstName: 'User', lastName: 'B', email: 'user.b@example.com', id: 2 }
        ];

        const results = await usersAdapter(mockData, { user: { id: 1 }, logger: mockLogger });

        expect(results).toBeDefined();
        expect(results).toMatchSnapshot();
    });
});
