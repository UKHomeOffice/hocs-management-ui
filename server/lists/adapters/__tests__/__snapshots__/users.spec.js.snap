// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Adapter should transform and sort user data 1`] = `
Array [
  Object {
    "label": "User B (user.b@example.com)",
    "value": 2,
  },
  Object {
    "label": "User C (user.c@example.com)",
    "value": 3,
  },
]
`;
